src/components/OnePerson.tsx

```diff
@@ -0,0 +1,71 @@
+import { Link, useParams } from 'react-router-dom';
+import { Person } from '../types';
+import classNames from 'classnames';
+
+type Props = {
+  person: Person;
+  people: Person[];
+  searchParams: URLSearchParams;
+};
+
+export const OnePerson: React.FC<Props> = ({
+  person,
+  people,
+  searchParams,
+}) => {
+  const { personSlug } = useParams();
+  const selectedPerson = personSlug ? personSlug : '';
```

Comment:
The `selectedPerson` logic should handle the absence of `personSlug` more meaningfully. Consider using a default value or handling the case where `personSlug` is undefined.